src/lib/util/corestr.cpp

```diff
@@ -171,15 +171,18 @@ void strreplacechr(std::string& str, char ch, char newch)
 	}
 }
 
-std::string strtrimspace(std::string& str)
+static std::string internal_strtrimspace(std::string& str, bool right_only)
 {
-	int start = 0;
-	for (auto & elem : str)
+	size_t start = 0;
+	if (!right_only)
 	{
-		if (!isspace(uint8_t(elem)))  break;
-		start++;
+		for (auto & elem : str)
```

Comment:
This is inefficient - you've got both an iterator and an index into the string in play here.  You only need one.